src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/ElementsSection.test.tsx

```diff
@@ -360,80 +364,77 @@ describe("ElementsSection allows card opening and closing", () => {
-    });
     const closeButton = getByTestId("close-element-card");
-    expect(closeButton).toBeInTheDocument();
-    act(() => {
-      fireEvent.click(closeButton);
-    });
+    await userEvent.click(closeButton);
+
+    expect(setSelectedDataElement).toHaveBeenCalledWith(null);
+    renderElementsSectionComponent(rerender, null);
     await waitFor(() => {
-      expect(screen.queryByTestId("data-element-card")).not.toBeInTheDocument();
+      expect(queryByTestId("data-element-card")).not.toBeInTheDocument();
     });
   });
 
-  test("should open and close a data element card manual close selection", async () => {
-    renderElementsSectionComponent();
+  test("should open and close a data element card with negation rationale", async () => {
+    const { rerender } = renderElementsSectionComponent(render, null);
     const elementSection = await findByTestId("elements-section");
-    expect(elementSection).toBeInTheDocument();
-    const adverseEventTab = screen.getByTestId("elements-tab-adverse_event");
-    expect(adverseEventTab).toBeInTheDocument();
-    act(() => {
-      userEvent.click(adverseEventTab);
-    });
+    const adverseEventTab = getByTestId("elements-tab-adverse_event");
+    userEvent.click(adverseEventTab);
     expect(adverseEventTab).toHaveAttribute("aria-selected", "true");
-    const adverseEventDataType = screen.getByTestId(
+    const adverseEventDataType = getByTestId(
       "data-type-Adverse Event: Encounter Inpatient"
     );
 
-    expect(adverseEventDataType).toBeInTheDocument();
-    act(() => {
-      fireEvent.click(adverseEventDataType);
-    });
+    userEvent.click(adverseEventDataType);
+    expect(setSelectedDataElement).toBeCalled();
+    selectedDataElement = testDataElements[11];
+
+    renderElementsSectionComponent(rerender, testDataElements[11]);
+    expect(getByTestId("data-element-card")).toBeInTheDocument();
 
-    await waitFor(() => {
-      expect(getByTestId("data-element-card")).toBeInTheDocument();
-    });
     await waitFor(() => {
       expect(queryByText("Negation Rationale")).not.toBeInTheDocument();
     });
-    const deviceTab = screen.getByTestId("elements-tab-device");
-    expect(deviceTab).toBeInTheDocument();
-    act(() => {
-      userEvent.click(deviceTab);
-    });
+    const deviceTab = getByTestId("elements-tab-device");
+
+    userEvent.click(deviceTab);
+
+    renderElementsSectionComponent(rerender, null);
     await waitFor(() => {
-      expect(screen.queryByTestId("data-element-card")).not.toBeInTheDocument();
+      expect(queryByTestId("data-element-card")).not.toBeInTheDocument();
     });
-    const deviceDataType = screen.getByTestId(
+    const deviceDataType = getByTestId(
       "data-type-Device, Order: Cardiopulmonary Arrest"
     );
-    act(() => {
-      fireEvent.click(deviceDataType);
-    });
+
+    userEvent.click(deviceDataType);
+    expect(setSelectedDataElement).toBeCalled();
+    selectedDataElement = testDataElements[14];
```

Comment:
I think this selectedDataElement might not be used. If so please delete

Reply:
I'll take care of this in MAT-6282